### PR TITLE
Vec3 basic array operations added

### PR DESCRIPTION
**Pull request by VKUDRI**
Basic vector 3x1 operations such as add/subtract/multiply/divide/dot/cross were added.

### CODE REVIEW

File: src/complex/Common/Array.hpp [hunk 1]
```diff
@@ -640,6 +641,63 @@ class Vec3 : public Array<T, 3>
...
+   */
+  inline T dot(const Vec3<T>& v) const
+  {
+    return (*this)[0] * v[0] + (*this)[1] * v[1] + (*this)[2] * v[2];
+  }
+
+  /**
+   * @brief Returns the magnitude of the 3x1 vector
+   */
+  inline T magnitude()
```
Comment (JDuffeyBQ) [inline src/complex/Common/Array.hpp:666]:
```suggestion
  inline T magnitude() const
```

File: src/complex/Common/Array.hpp [hunk 2]
```diff
@@ -640,6 +641,63 @@ class Vec3 : public Array<T, 3>
...
+  inline Vec3& operator/=(const T& r)
+  {
+    (*this)[0] /= r, (*this)[1] /= r, (*this)[2] /= r;
+    return *this;
+  }
+  inline Vec3& operator*=(const T& r)
+  {
+    (*this)[0] *= r, (*this)[1] *= r, (*this)[2] *= r;
+    return *this;
+  }
```
Comment (JDuffeyBQ) [inline src/complex/Common/Array.hpp]: Split comma operator usage into multiple lines.

Comment (JDuffeyBQ) [inline src/complex/Common/Array.hpp]: For primitive type `T`, pass by value.

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]
How much are we doing with the matrix class? I'd want to avoid turning it into Eigen.

File: src/complex/Common/Array.hpp (hunk 1 above)
Comment (imikejackson) [reply src/complex/Common/Array.hpp:666]: There are some very basic vector3 operations that complex/dream3d tend to do and this is one of them so I'll let this through.

File: src/complex/Common/Array.hpp (hunk 2 above)
Comment (imikejackson) [reply src/complex/Common/Array.hpp]: Done

Comment (imikejackson) [reply src/complex/Common/Array.hpp]: Done

**Review by imikejackson — APPROVED** [review]

**Comment by JDuffeyBQ** [comment]
Are we keeping the matrix classes?

**Pull request merged**